lib/alephant/broker/load_strategy/revalidate/refresher.rb

```diff
@@ -0,0 +1,62 @@
+module Alephant
+  module Broker
+    module LoadStrategy
+      module Revalidate
+        class Refresher
+          include Logger
+
+          attr_reader :component_meta
+
+          def initialize(component_meta)
+            @component_meta = component_meta
+          end
+
+          def refresh
+            return if cache.get(inflight_cache_key)
+
+            queue.send_message(component_meta.to_json)
+            cache.set(inflight_cache_key, true)
+          end
+
+          private
+
+          def queue
+            @queue ||= begin
```

Comment:
Heh, must be different programming circles - i've (and everyone i've previously worked with) always used a `begin`/`end` block for memoization.  Never even given a thought to using a `Proc` or `lambda`.  Hmm, wonder if there is a performance benefit/trade-off to either approach?